library/src/schemas/variant/variant.ts

```
@@ -19,6 +19,67 @@ import type {
+    ) {
+      // @ts-expect-error
+      values = discriminatorSchema.options;
+    } else {
+      // Non-enumerable discriminator (e.g. optional, union, custom)
+      return null;
+    }
+    for (const value of values) {
+      // `NaN` is excluded because `Map` lookups use SameValueZero (NaN matches
+      // NaN) whereas a `literal` discriminator compares with `===` (NaN never
+      // matches), so a NaN discriminator must use the slow path for parity
+      if (value !== value) {
```

Biome (2.5.1)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)